src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java

```diff
@@ -285,6 +323,49 @@ public String getQuestionResultStatisticsCsv(
         return normalisedRankOfResponse;
     }
 
+    /**
+     * Returns list of responses excluding responses given to self.
+     *
+     * @param responses a list of responses
+     * @return list of responses excluding self given responses
+     */
+    private List<FeedbackResponseAttributes> getResponsesExcludingSelf(List<FeedbackResponseAttributes> responses) {
+        List<FeedbackResponseAttributes> responsesExcludingSelf = new ArrayList<FeedbackResponseAttributes>();
+        for (FeedbackResponseAttributes response : responses) {
+            if (!response.giver.equalsIgnoreCase(response.recipient)) {
+                responsesExcludingSelf.add(response);
+            }
+        }
+        return responsesExcludingSelf;
+    }
+
+    private String getAverageExcludingSelfText(boolean showAvgExcludingSelf, DecimalFormat df, Double averageExcludingSelf) {
+        if (showAvgExcludingSelf) {
+            // Display a dash if the user has only self response
+            return averageExcludingSelf == null ? "-" : df.format(averageExcludingSelf);
+        }
+        return "";
+    }
+
+    /**
+     * Returns map of recipient ranks excluding self.
+     */
+    private Map<String, List<Integer>> getRecipientRanksExcludingSelf(FeedbackQuestionAttributes question,
+            List<FeedbackResponseAttributes> responses) {
+        Map<String, List<Integer>> recipientRanksExcludingSelf = null;
```

Comment:
Maybe introduce some method that checks for the condition? That way the code becomes self-documenting.